contracts/ExitFormat.sol

```diff
@@ -28,15 +28,18 @@ library ExitFormat {
     // (which would make a material difference to the final state in the case of running out of gas or funds)
     // Allocations = Allocation[]
 
+    enum AllocationType {simple, withdrawHelper, guarantee}
+
     // An Allocation specifies
     // * a destination, referring either to an ethereum address or an application-specific identifier
     // * an amount of asset
+    // * an allocationType, which directs calling code on how to interpret the allocation
     // * custom metadata (optional field, can be zero bytes). This can be used flexibly by different protocols.
     struct Allocation {
         bytes32 destination;
         uint256 amount;
-        address callTo; // compatible with Vetor WithdrawHelper
-        bytes metadata; // compatible with Vetor WithdrawHelper
+        uint8 allocationType;
```

Comment:
This being a `uint8` rather than an `AllocationType` came up in discussion at some point. I don't remember if there's a good rationale for it. The plan for it to be an int may have predated the intention to explicitly name the allocation types in an enum, and it may no longer be a good idea.

Reply:
I think the idea was that having it as an `AllocationType` ties the format to that particular enum. Having it as a `uint8` would allow different enums to be used (e.g. by other projects), and breaks the whole thing into two separate standards . Does that make sense?